src/App.tsx

```diff
@@ -26,70 +29,121 @@ type ReorderOptions = {
   isReversed: boolean,
 };
 
-// Use this function in the render to prepare goods
 export function getReorderedGoods(
   goods: string[],
   { sortType, isReversed }: ReorderOptions,
 ) {
-  // To avoid the original array mutation
-  const visibleGoods = [...goods];
+  let visibleGoods = [...goods];
 
-  // Sort and reverse goods if needed
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+  switch (sortType) {
+    case SortType.LENGTH:
+      visibleGoods.sort((good1, good2) => {
+        return good1.length - good2.length;
+      });
+      break;
+
+    case SortType.ALPABET:
+      visibleGoods.sort((good1, good2) => {
+        return good1.localeCompare(good2);
+      });
+      break;
+
+    case SortType.NONE:
+      visibleGoods = [...goodsFromServer];
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
-
-      <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
-      </ul>
-    </div>
-  );
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
 };
+
+export class App extends React.Component<{}, State> {
+  state = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  alphabetSort() {
+    this.setState({
+      sortType: SortType.ALPABET,
+    });
+  }
+
+  sortByLength() {
+    this.setState({
+      sortType: SortType.LENGTH,
+    });
+  }
+
+  reverseArr() {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  }
+
+  resetArr() {
+    this.setState({
+      sortType: SortType.NONE,
+      isReversed: false,
+    });
+  }
```

Comment:
that's not required but it would be good to combine into a single function with switch/case